actionmailbox/app/controllers/action_mailbox/ingresses/mailgun/inbound_emails_controller.rb

```diff
@@ -64,14 +64,22 @@ def authenticated?
           ).authenticated?
         else
           raise ArgumentError, <<~MESSAGE.squish
-            Missing required Mailgun API key. Set action_mailbox.mailgun_api_key in your application's
-            encrypted credentials or provide the MAILGUN_INGRESS_API_KEY environment variable.
+            Missing required Mailgun Signing key. Set action_mailbox.mailgun_signing_key in your application's
+            encrypted credentials or provide the MAILGUN_INGRESS_SIGNING_KEY environment variable.
           MESSAGE
         end
       end
 
       def key
-        Rails.application.credentials.dig(:action_mailbox, :mailgun_api_key) || ENV["MAILGUN_INGRESS_API_KEY"]
+        if Rails.application.credentials.dig(:action_mailbox, :mailgun_api_key) ||  ENV["MAILGUN_INGRESS_API_KEY"]
```

Comment:
We could be more specific about which deprecated configuration method is being used:

```ruby
if Rails.application.credentials.dig(:action_mailbox, :mailgun_api_key)
  ActiveSupport::Deprecation.warn(<<-MSG.squish)
    Rails.application.credentials.action_mailbox.api_key is deprecated and will be ignored in Rails 6.2.
    Use Rails.application.credentials.action_mailbox.signing_key instead.
  MSG
  Rails.application.credentials.dig(:action_mailbox, :mailgun_api_key)
elsif ENV["MAILGUN_INGRESS_API_KEY"]
  ActiveSupport::Deprecation.warn(<<-MSG.squish)
    The MAILGUN_INGRESS_API_KEY environment variable is deprecated and will be ignored in Rails 6.2.
    Use MAILGUN_INGRESS_SIGNING_KEY instead.
  MSG
  ENV["MAILGUN_INGRESS_API_KEY"]
else
  Rails.application.credentials.dig(:action_mailbox, :mailgun_signing_key) || ENV["MAILGUN_INGRESS_SIGNING_KEY"]
end
```